bookmarks.go

```diff
@@ -0,0 +1,159 @@
+package slack
+
+import (
+	"context"
+	"net/url"
+)
+
+type Bookmark struct {
+	ID        string   `json:"id"`
+	ChannelID string   `json:"channel_id"`
+	Title     string   `json:"title"`
+	Link      string   `json:"link"`
+	Emoji     string   `json:"emoji"`
+	IconURL   string   `json:"icon_url"`
+	Type      string   `json:"type"`
+	Created   JSONTime `json:"date_created"`
+	Updated   JSONTime `json:"date_updated"`
+	Rank      string   `json:"rank"`
+
+	LastUpdatedByUserID string `json:"last_updated_by_user_id"`
+	LastUpdatedByTeamID string `json:"last_updated_by_team_id"`
+
+	ShortcutID string `json:"shortcut_id"`
+	EntityID   string `json:"entity_id"`
+	AppID      string `json:"app_id"`
+}
+
+type AddBookmarkParameters struct {
+	Title    string // A required title for the bookmark
+	Type     string // A required type for the bookmark
+	Link     string // URL required for type:link
+	Emoji    string // An optional emoji
+	EntityID string
+	ParentID string
+}
+
+type EditBookmarkParameters struct {
+	Title *string // Change the title. Set to "" to clear
+	Emoji *string // Change the emoji. Set to "" to clear
+	Link  string  // Change the link
+}
+
+type addBookmarkReponse struct {
```

Comment:
👀 
```suggestion
type addBookmarkResponse struct {
```

Reply:
🤦 this is what I get for not enabling spell check in vs code, will fix. thanks!